compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/SchemaGenerator.java

```diff
@@ -152,6 +164,35 @@ private void findRootTypes(ServiceDeclarationSymbol serviceDeclarationSymbol) {
         }
     }
 
+    private Collection<? extends MethodSymbol> getMethods(Node node) {
+        if (node.kind() == SyntaxKind.SERVICE_DECLARATION) {
+            return getMethods((ServiceDeclarationNode) node);
+        }
+
+        if (node.kind() == SyntaxKind.OBJECT_CONSTRUCTOR) {
+            return getMethods((ObjectConstructorExpressionNode) node);
+        }
+
+        return new ArrayList<>();
+    }
+
+    private Collection<? extends MethodSymbol> getMethods(
+            ObjectConstructorExpressionNode objectConstructorExpressionNode) {
+        // noinspection OptionalGetWithoutIsPresent
+        return objectConstructorExpressionNode.members().stream()
+                .filter(member -> member instanceof FunctionDefinitionNode && context.semanticModel().symbol(member)
+                        .isPresent()).map(methodNode -> (MethodSymbol) context.semanticModel().symbol(methodNode).get())
```

Comment:
Let's avoid using `instanceof` checks as much as possible. The SymbolKind and TypeKind are there to avoid these checks and to help downcasting.

Reply:
Updated